### PR TITLE
add kstem token filter docs #8150

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
add kstem token filter docs

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/8150

### Version
all

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
@udabhas @varun-lodaya Please provide tech review approval to move this PR forward in the documentation process. Please review this week or provide a peer who can review it. Thank you.

File: _analyzers/token-filters/index.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ Token filter | Underlying Lucene token filter|  Description
...
 `keyword_marker` | [KeywordMarkerFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/KeywordMarkerFilter.html) | Marks specified tokens as keywords, preventing them from being stemmed.
 `keyword_repeat` | [KeywordRepeatFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/KeywordRepeatFilter.html) | Emits each incoming token twice: once as a keyword and once as a non-keyword.
-`kstem` | [KStemFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/en/KStemFilter.html) | Provides kstem-based stemming for the English language. Combines algorithmic stemming with a built-in dictionary.
+[`kstem`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/kstem/) | [KStemFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/en/KStemFilter.html) | Provides kstem-based stemming for the English language. Combines algorithmic stemming with a built-in dictionary.
```
Comment (natebower) [inline _analyzers/token-filters/index.md:37]:
```suggestion
[`kstem`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/kstem/) | [KStemFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/en/KStemFilter.html) | Provides KStem-based stemming for the English language. Combines algorithmic stemming with a built-in dictionary.
```

File: _analyzers/token-filters/kstem.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+---
+layout: default
+title: Kstem
```
Comment (natebower) [inline _analyzers/token-filters/kstem.md:3]:
```suggestion
title: KStem
```

File: _analyzers/token-filters/kstem.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+parent: Token filters
+nav_order: 220
+---
+
+# KStem token filter
+
+The `kstem` token filter is a stemming filter used to reduce words to their root forms. The filter is a lightweight algorithmic stemmer designed for the English language that performs the following stemming:
```
Comment (natebower) [inline _analyzers/token-filters/kstem.md:10]:
```suggestion
The `kstem` token filter is a stemming filter used to reduce words to their root forms. The filter is a lightweight algorithmic stemmer designed for the English language that performs the following stemming operations:
```

File: _analyzers/token-filters/kstem.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+- Reduces plurals to singular form.
```
Comment (natebower) [inline _analyzers/token-filters/kstem.md:12]:
```suggestion
- Reduces plurals to their singular form.
```

File: _analyzers/token-filters/kstem.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+- Converts different tenses of verbs to their base form.
```
Comment (natebower) [inline _analyzers/token-filters/kstem.md:13]:
```suggestion
- Converts different verb tenses to their base form.
```

File: _analyzers/token-filters/kstem.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+- Removes common derivational endings such as "-ing", "-ed".
```
Comment (natebower) [inline _analyzers/token-filters/kstem.md:14]:
```suggestion
- Removes common derivational endings, such as "-ing" or "-ed".
```

File: _analyzers/token-filters/kstem.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+The `kstem` token filter is equivalent to the a `stemmer` filter configured with a `light_english` language. It provides a more conservative stemming compared to other stemming filters like `porter_stem`.
```
Comment (natebower) [inline _analyzers/token-filters/kstem.md:16]:
```suggestion
The `kstem` token filter is equivalent to the a `stemmer` filter configured with a `light_english` language. It provides more conservative stemming compared to other stemming filters, like `porter_stem`.
```

File: _analyzers/token-filters/kstem.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
```
Comment (natebower) [inline _analyzers/token-filters/kstem.md:17]: Line 16: Remove either "a" or "the" preceding `stemmer`.

File: _analyzers/token-filters/kstem.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+The `kstem` token filter is based on the Lucene KStemFilter. For more information, see the [Lucene documentation](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/en/KStemFilter.html).
+
+## Example
+
+The following example request creates a new index named `my_kstem_index` and configures an analyzer with `kstem` filter:
```
Comment (natebower) [inline _analyzers/token-filters/kstem.md:22]:
```suggestion
The following example request creates a new index named `my_kstem_index` and configures an analyzer with a `kstem` filter:
```

File: _analyzers/token-filters/kstem.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Generated tokens
+
+Use the following request to examine the tokens generated using the created analyzer:
```
Comment (natebower) [inline _analyzers/token-filters/kstem.md:60]:
```suggestion
Use the following request to examine the tokens generated using the analyzer:
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my changes and let me know if you have any questions. Thanks!

**Review by kolchfa-aws — APPROVED** [review]

**Pull request merged**